crates/modelardb_manager/src/remote.rs

```diff
@@ -447,21 +447,23 @@ impl FlightService for FlightServiceHandler {
             let server_mode = ServerMode::from_str(mode).map_err(Status::invalid_argument)?;
             let node = Node::new(url.to_string(), server_mode.clone());
 
-            // Use the metadata manager to persist the node to the metadata Delta Lake.
+            // Use the cluster to register the node in memory. This returns an error if the node is
+            // already registered.
             self.context
-                .remote_data_folder
-                .metadata_manager
-                .save_node(&node)
+                .cluster
+                .write()
                 .await
+                .register_node(node.clone())
                 .map_err(|error| Status::internal(error.to_string()))?;
 
-            // Use the cluster to register the node in memory. Note that if this fails, the cluster
-            // and metadata Delta Lake will be out of sync until the manager is restarted.
+            // Use the metadata manager to persist the node to the metadata Delta Lake. Note that if
```

Comment:
Could the problem of the metadata Delta Lake and the manager being out of sync be remedied if the code checked if `save_node()` returned an error and deleted `node` from `cluster` before returning the error? Of course that leaves a small window where `node` could be used by `modelardbm`, but it seems better than leaving it registered. For a completely safe implementation a transactional system with a staging area is probably needed so changes can be staged until all involved have verified that they can perform the required task before all committing, but this seems out of scope for this PR as this should probably be done in a separate component used throughout the system due to how crucial it is to implement correctly and how easy it is to make mistakes.